File: backend/app/schemas/company.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class CompanyCreate(BaseModel):
    name: str
    address: str | None = None
    gst_no: str | None = None
    pan_no: str | None = None
    logo_url: str | None = None
    is_default: bool = False
    phone: str | None = None
    email: str | None = None
    bank_name: str | None = None
    account_no: str | None = None
    ifsc_code: str | None = None
    branch: str | None = None


class CompanyUpdate(BaseModel):
    name: str | None = None
    address: str | None = None
    gst_no: str | None = None
    pan_no: str | None = None
    logo_url: str | None = None
    is_default: bool | None = None
    phone: str | None = None
    email: str | None = None
    bank_name: str | None = None
    account_no: str | None = None
    ifsc_code: str | None = None
    branch: str | None = None


class CompanyResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    name: str
    address: str | None
    gst_no: str | None
    pan_no: str | None
    logo_url: str | None
    is_default: bool
    phone: str | None
    email: str | None
    bank_name: str | None
    account_no: str | None
    ifsc_code: str | None
    branch: str | None
    created_at: datetime
    updated_at: datetime

File: backend-spring/src/main/java/com/nearkart/entity/Company.java
package com.nearkart.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "companies")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
public class Company {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String address;

    @Column(name = "gst_no")
    private String gstNo;

    @Column(name = "pan_no")
    private String panNo;

    @Column(name = "logo_url")
    private String logoUrl;

    @Column(name = "is_default")
    @Builder.Default
    private Boolean isDefault = false;

    private String phone;
    private String email;

    @Column(name = "bank_name")
    private String bankName;

    @Column(name = "account_no")
    private String accountNo;

    @Column(name = "ifsc_code")
    private String ifscCode;

    private String branch;

    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt;

    @PrePersist
    protected void onCreate() { OffsetDateTime now = OffsetDateTime.now(); createdAt = now; updatedAt = now; }

    @PreUpdate
    protected void onUpdate() { updatedAt = OffsetDateTime.now(); }
}
